### PR TITLE
Revert "Revert "WebUser restore""

### PR DESCRIPTION
**Pull request by czue**
Reverts dimagi/commcare-hq#11890

don't merge until http://manage.dimagi.com/default.asp?228568 is fixed on this branch

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 1]
```diff
@@ -22,54 +24,179 @@
...
+        both the toggle is enabled for the domain, and the user has a non-zero,
+        non-null factor set.
+        """
+        if ENABLE_LOADTEST_USERS.enabled(self.domain):
+            return self._loadtest_factor or 1
+        return 1
+
+    @property
+    def username(self):
+        return self._couch_user.username
```
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: I think this should be `self._couch_user.raw_username` to fix the bug

Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: sounds right - I wasn't going to change it until I added a test

File: corehq/ex-submodules/casexml/apps/phone/tests/test_ota_restore.py [hunk 2]
```diff
@@ -130,26 +139,32 @@ def testUserRestoreWithCase(self):
...
+            xml.get_case_xml(
+                newcase,
+                [case_const.CASE_ACTION_CREATE, case_const.CASE_ACTION_UPDATE]
+            )
+        )
 
         # check v2
         expected_v2_case_block = """
-        <case case_id="asdf" date_modified="2010-06-29T13:42:50.000000Z" user_id="foo" xmlns="http://commcarehq.org/case/transaction/v2" >
+        <case case_id="asdf" date_modified="2010-06-29T13:42:50.000000Z" user_id="{user_id}" xmlns="http://commcarehq.org/case/transaction/v2" >
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_ota_restore.py]: E501 line too long (144 > 115 characters)

File: corehq/apps/hqcase/tests/test_case_sharing.py [hunk 3]
```diff
@@ -60,9 +60,9 @@ def update_and_test(case_id, owner=None, should_have=None, should_not_have=None)
 
         def check_has_block(case_block, should_have, should_not_have, line_by_line=True):
             for user in should_have:
-                check_user_has_case(self, user.to_casexml_user(), case_block, line_by_line=line_by_line)
+                check_user_has_case(self, user.to_ota_restore_user(), case_block, line_by_line=line_by_line)
             for user in should_not_have:
-                check_user_has_case(self, user.to_casexml_user(), case_block, should_have=False, line_by_line=line_by_line)
+                check_user_has_case(self, user.to_ota_restore_user(), case_block, should_have=False, line_by_line=line_by_line)
```
Comment (dimagimon) [inline corehq/apps/hqcase/tests/test_case_sharing.py]: E501 line too long (127 > 115 characters)

**Comment by czue** [comment]
this should be good to go when :white_check_mark:

**Comment by snopoke** [comment]
build timed out - try merge master

**Comment by czue** [comment]
:white_check_mark:

**Comment by czue** [comment]
@calellowitz @emord fyi, take two on the restore update going out today. maybe you can deploy a little earlier in the day so there's time to confirm nothing has gone wrong?

**Comment by benrudolph** [comment]
nice! thanks guys. this is so cool!

**Pull request merged**